tests/image/classification/test_data.py

```diff
@@ -221,6 +221,40 @@ def test_from_filepaths_visualise(tmpdir):
     dm.show_train_batch(["per_sample_transform", "per_batch_transform"])
 
 
+@pytest.mark.skipif(not _IMAGE_AVAILABLE, reason="image libraries aren't installed.")
+@pytest.mark.skipif(not _MATPLOTLIB_AVAILABLE, reason="matplotlib isn't installed.")
+def test_from_filepaths_visualise_subplots_exceding_max_cols(tmpdir):
+    tmpdir = Path(tmpdir)
+
+    (tmpdir / "e").mkdir()
+    _rand_image().save(tmpdir / "e_1.png")
+
+    train_images = [
+        str(tmpdir / "e_1.png"),
+    ] * 8
```

Comment:
We were not able to catch this bug because of a missing text having a number of images > max_cols * N ( N > 1 ). Here, the number of images we pass is 8, and hence the subplot will have 4 rows and 2 columns.

Also ensured that this test fails on master (without the fix in this PR).